packages/pages/src/App.js

```diff
@@ -4,46 +4,75 @@ import { format, parse } from "json-rpc-protocol";
 
 const App = ({ effects, state }) => (
   <div>
-    <div name="listEntries">
-      <button type="button" onClick={effects.refreshEntries}>
-        Refresh
-      </button>
-      <ul>
-        {state.entries.map(entry => (
-          <li key={entry.id}>
-            {entry.name} {entry.content}
-            <button
-              type="button"
-              value={entry.id}
-              onClick={effects.deleteEntry}
-            >
-              Delete
+    {state.id !== "" ? (
+      <form onSubmit={effects.submitUpdates}>
+        <label>
+          <input type="text" value={state.name} onChange={effects.changeName} />
+        </label>
+        <label>
+          <input
+            type="text"
+            value={state.content}
+            onChange={effects.changeContent}
+          />
+        </label>
+        <input type="submit" value="Submit" />
+      </form>
+    ) : (
+      <div>
+        <button type="button" onClick={effects.refreshEntries}>
+          Refresh
+        </button>
+        <ul>
+          {state.entries.map(entry => (
+            <li key={entry.id}>
+              {entry.name} {entry.content}
+              <button
+                type="button"
+                value={entry.id}
+                onClick={effects.deleteEntry}
+              >
+                Delete
+              </button>
+              <button
+                type="button"
+                onClick={effects.updateEntry.bind(
+                  this,
+                  entry.id,
+                  entry.name,
+                  entry.content
+                )}
```

Comment:
Avoid creating functions (that's what `bind` does) in the rendering function, you can use [`data-*`](https://devdocs.io/html/attributes#data-attribute) to associate value with the `<button>`.

Even in this case, you should probably only attach the `id` and fetch the rest of the data directly in the state.